src/main/java/org/jenkinsci/plugins/electricflow/ElectricFlowGlobalConfiguration.java

```diff
@@ -35,7 +37,6 @@ public ElectricFlowGlobalConfiguration() {
 
   @Override
   public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
-    this.configurations = null;
```

Comment:
Are we sure this change is safe? While most of the other changes look fine to me, I'm worried this one might be introducing a regression. Have you tested it? It seems to me that it might be important to clear list when saving the form, for example to ensure that removing the last entry of the list actually persists to disk. If there is any doubt here, I would rather defer such cleanup to a separate PR.